$MirOS: ports/mailnews/pine/patches/patch-pine_reply_c,v 1.1.7.1 2005/03/18 15:49:48 tg Exp $
--- pine/reply.c.orig	Thu Apr 28 17:21:41 2005
+++ pine/reply.c	Wed May  4 23:28:12 2005
@@ -4501,10 +4501,10 @@ bounce(pine_state, role)
 
     if(mn_total_cur(pine_state->msgmap) > 1L){
 	save_toptr = &save_to;
-	sprintf(tmp_20k_buf, "BOUNCE (redirect) %d messages to : ",
+	sprintf(tmp_20k_buf, "BOUNCE (redirect) %ld messages to : ",
 		mn_total_cur(pine_state->msgmap));
 	prmpt_who = cpystr(tmp_20k_buf);
-	sprintf(tmp_20k_buf, "Send %d messages ",
+	sprintf(tmp_20k_buf, "Send %ld messages ",
 		mn_total_cur(pine_state->msgmap));
 	prmpt_cnf = cpystr(tmp_20k_buf);
     }
@@ -5457,7 +5457,8 @@ generate_message_id()
       hostpart = cpystr("huh");
 
     sprintf(id,"<Pine.%.4s.%.20s.%02d%02d%02d%02d%02d%02d%X.%d@%.50s>",
-	    SYSTYPE, pine_version, (now_x->tm_year) % 100, now_x->tm_mon + 1,
+	    SYSTYPE, pine_version, (int)((now_x->tm_year) % 100),
+	    now_x->tm_mon + 1,
 	    now_x->tm_mday, now_x->tm_hour, now_x->tm_min, now_x->tm_sec, 
 	    cnt, getpid(), hostpart);
 
